### PR TITLE
Compare different Agents by using IPS

### PR DESCRIPTION
**Pull request by ihtiihti**
This set of changes introduces the following features:
* Gathering Probability Scores from different Agents
* Launch agents in environments and collect logs
* Calculate Inversed Propensity Scores
* Compare different agents under the same conditions

The changes are the very first step for offline A/B Testing of Reco Gym

### CODE REVIEW

**Comment by drohde** [comment]
@emmanuelguerin @flavianv

**Comment by emmanuelguerin** [comment]
@drohde Did you mean Emmanuel Augustin? I don't know his github handle, though.

**Comment by drohde** [comment]
> @drohde Did you mean Emmanuel Augustin? I don't know his github handle, though.

yes, sorry @emmanuelguerin

File: reco_gym/envs/abstract.py [hunk 1]
```diff
@@ -187,35 +229,47 @@ def step_offline2(self):
...
-                    pd.DataFrame().from_dict(
-                        {'v': -1, 'u': user_id, 'r': [action], 'c': reward}))
+                if observations is not None:
+                    for observation in observations:
+                        data['v'].append(observation[1])
+                        data['u'].append(user_id)
+                        data['r'].append(-1)
+                        data['c'].append(-1)
+                        data['ps'].append(None)
+
```
Comment (drohde) [inline reco_gym/envs/abstract.py]: It wouldn't hurt to add:
data['z']='organic' ... etc...

**Comment by drohde** [comment]
@emxcagt

**Comment by ihtiihti** [comment]
A new pull request will be provided with all recommended fixes.

**Pull request closed without merging**